carp.common/src/commonTest/kotlin/dk/cachet/carp/common/application/devices/WebBrowserTest.kt

```diff
@@ -0,0 +1,57 @@
+package dk.cachet.carp.common.application.devices
+
+import dk.cachet.carp.common.application.Trilean
+import dk.cachet.carp.common.application.URL
+import dk.cachet.carp.common.infrastructure.serialization.createDefaultJSON
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class WebBrowserTest
+{
+    @Test
+    fun registration_builder_sets_properties()
+    {
+        val registration = WebBrowserDeviceRegistrationBuilder().apply {
+            browser = "CHROME"
+            url = "https://www.example.com"
+        }.build()
+
+        assertEquals("CHROME", registration.browser)
+        assertEquals("https://www.example.com", registration.url.stringRepresentation)
+    }
+
+    @Test
+    fun registration_browser_and_uri_is_serialized()
```

Comment:
I think I added tests for some of the early types as regression tests to double-check that base properties were serialized. To a large extent, I think these existing tests suffice, and from this point onwards, we can rely on the serialization framework to work as expected.

I would thus suggest removing this; otherwise we'd have to add this fairly boiler-plate code (which I'm fairly certain provides no extra coverage) to each extending type.

To make that clear, probably the existing type-specific tests should somehow convey they are more "generic" in the nature of what they test. :) You can ignore that for now.